libyara/modules/dex.c

```diff
@@ -405,7 +405,7 @@ uint32_t load_encoded_field(
       encoded_field.access_flags);
   #endif
 
-  uint64_t name_idx = get_integer(
+  int name_idx = get_integer(
```

Comment:
I see, `name_idx` is being used as the argument for `%i`, but we should change the format string instead of the type for the variable, because otherwise we are going to get some other warnings due to loss of precision in the conversion from `int64_t` to `int` in 32-bits platforms.

Reply:
I figured that those loop counters should always fit into 31 bits... The alternative was replacing `%d` with `%"PRIu64"`. Which is fine by me, too.

Reply:
You are probably right about the integer fitting into 31bits, using an `int64_t` is probably an overkill. We can use `int` but have an explicit typecast, I think that should be enough for preventing the warnings due to loss of precision.